pkg/interceptors/webhook/sink.go

```diff
@@ -50,27 +50,28 @@ func NewInterceptor(wh *triggersv1.WebhookInterceptor, c *http.Client, ns string
 	}
 }
 
-func (w *Interceptor) ExecuteTrigger(payload []byte, request *http.Request, trigger *triggersv1.EventListenerTrigger, eventID string) ([]byte, error) {
+func (w *Interceptor) ExecuteTrigger(payload []byte, request *http.Request, trigger *triggersv1.EventListenerTrigger, eventID string) ([]byte, http.Header, error) {
```

Comment:
The Interceptor interface seems very strange with regard for the method signature `ExecuteTrigger(payload []byte, request *http.Request, trigger *triggersv1.EventListenerTrigger, eventID string) ([]byte, error)`. The passed trigger is never used and since we only have validator interceptors (no decorators), the payload is always returned as is. I feel like ```type Interceptor interface {
	Intercept(req *http.Request) (*http.Response, error)
}``` might make more sense, where the payload and headers should be passed within the response. Any initialization/decoration of the original event request would be done before the interceptor chain (e.g. add eventID/triggers headers). Doesn't need to be done here, but just as a followup. @dibyom @ncskier @wlynch

Reply:
See #269 which is similar

Reply:
Right, I remember now Billy mentioned on the call.